mobile/src/screens/LoadingScreen.tsx

```diff
@@ -0,0 +1,56 @@
+import { useNavigation } from "@react-navigation/native";
+import React, { useContext, useState } from "react";
+import { Text, View, Image } from "react-native";
+
+import stemettImage from "../../assets/Stemett.png";
+import Button from "../components/Button";
+import TwoButtonPopup from "../components/popups/TwoButtonPopup";
+import { SocketContext } from "../contexts/SocketContext";
+import { UserContext } from "../contexts/UserContext";
+import { AppStylesheet } from "../styles/AppStylesheet";
+
+const LoadingScreen = () => {
+  const navigation = useNavigation();
+  const socket = useContext(SocketContext);
+  const { userState } = useContext(UserContext);
+  const [stopPopup, setStopPopup] = useState(false);
+  let playerColor = "";
+
+  const quitSearch = () => {
+    setStopPopup(false);
+    navigation.navigate("HomeScreen");
+    socket.emit("quit searching");
+  };
+
+  socket.once("successful assign", (color: string) => {
+    playerColor = color;
+  });
+
+  socket.once("game ready", () => {
+    navigation.navigate("Chess", { color: playerColor });
```

Comment:
My understanding was that by creating a type for the navigation, we can specify a type here. For instance how it is done in [this](https://github.com/wcandillon/can-it-be-done-in-react-native/blob/07c223e8d6ff2bd1f11afc69e0e0ad9af40c31ca/season4/App.tsx#L43) project. This is why originally the type was used in the declaration of the stack navigator. I'm not sure why that was changed or if I'm correct in assuming this. Please let me know.